server/routes/charts.js

```diff
@@ -71,6 +73,17 @@ router.route('/charts/:chart_id').get(ensureAuthenticated, async (req, res) => {
       studyTotals,
       filters,
     }
+    if (req.headers['content-type'] === 'text/csv') {
+      res.header('Content-Type', 'text/csv')
+
+      // await csvDownloader(res, graphTable)
+
+      // return res.send()
+      const csvService = new CsvService(graphTable)
+
+      await csvService.toCsv(res)
```

Comment:
it feels weird to pass the response object to the csv service. can the service return the csv data and then we handle putting that in the response here?

Reply:
https://github.com/AMP-SCZ/dpdash/pull/287/commits/ac482a8d76ec55f2853a409d3c4b2d18c84ada34